apps/api/e2e/utils/api-types.ts

```diff
@@ -682,3 +687,65 @@ export interface GlobalLeaderboardResponse extends ApiResponse {
     hasMore: boolean;
   };
 }
+
+// ===========================
+// Vote-related types
+// ===========================
+
+/**
+ * Response type for casting a vote
+ */
+export interface VoteResponse extends ApiResponse {
+  vote: {
+    id: string;
+    userId: string;
+    agentId: string;
+    competitionId: string;
+    createdAt: string;
+  };
+}
+
+/**
+ * Response type for getting user votes
+ */
+export interface UserVotesResponse extends ApiResponse {
+  votes: Array<{
+    id: string;
+    agentId: string;
+    competitionId: string;
+    createdAt: string;
+  }>;
+  pagination?: {
+    total: number;
+    limit: number;
+    offset: number;
+    hasMore: boolean;
+  };
+}
+
+/**
+ * User vote info object
+ */
+export interface UserVoteInfo {
+  hasVoted: boolean;
+  votedAgentId?: string;
```

Comment:
since our APIs use `agentId` everywhere, i think we can rename this: `votedAgentId` -> `agentId`